lib/features/home/bloc/location_bloc.dart

```diff
@@ -12,18 +14,24 @@ const String _dateFormat = 'dd-MMM-yyyy';
 
 class LocationBloc extends Cubit<LocationState> {
   final LocationService _locationService;
+  final GeocoderService _geocoderService;
 
-  LocationBloc({LocationService? locationService})
+  LocationBloc(
+      {LocationService? locationService, GeocoderService? geocoderService})
       : _locationService = locationService ?? LocationService(),
+        _geocoderService = geocoderService ?? GeocoderService(),
         super(LocationInitial(PermissionStatusDiary.denied));
 
   Future<void> requestCurrentLocation() async {
+    final String dateDisplay = DateFormat(_dateFormat).format(DateTime.now());
     try {
       final LocationDetails data = await _locationService.getCurrentLocation();
-      final String dateDisplay = DateFormat(_dateFormat).format(DateTime.now());
-
-      emit(LocationReadyState(
-          LatLng(data.latitude, data.longitude), dateDisplay));
+      await _geocoderService
```

Comment:
`await` and `then` at the same time? not wrong, but not a good practice

Reply:
i will choose one of them